torchvision/csrc/cpu/image/readpng_cpu.cpp

```diff
@@ -72,30 +74,74 @@ torch::Tensor decodePNG(const torch::Tensor& data) {
     TORCH_CHECK(retval == 1, "Could read image metadata from content.")
   }
 
-  int channels;
-  switch (color_type) {
-    case PNG_COLOR_TYPE_RGB:
-      channels = 3;
-      break;
-    case PNG_COLOR_TYPE_RGB_ALPHA:
-      channels = 4;
-      break;
-    case PNG_COLOR_TYPE_GRAY:
-      channels = 1;
-      break;
-    case PNG_COLOR_TYPE_GRAY_ALPHA:
-      channels = 2;
-      break;
-    case PNG_COLOR_TYPE_PALETTE:
-      channels = 1;
-      break;
-    default:
-      png_destroy_read_struct(&png_ptr, &info_ptr, nullptr);
-      TORCH_CHECK(false, "Image color type is not supported.");
+  int current_channels = png_get_channels(png_ptr, info_ptr);
+
+  if (channels > 0 && channels != current_channels) {
+    // TODO: consider supporting PNG_INFO_tRNS
```

Comment:
There is another type of transparency that we can support. Not sure if it's a must-have for this PR.

Reply:
No need to add this for now I believe